improver/calibration/reliability_calibration.py

```diff
@@ -585,6 +585,87 @@ def _ensure_monotonicity(self, cube):
             warnings.warn(msg)
             cube.data = np.sort(cube.data, axis=threshold_dim)
 
+    @staticmethod
+    def _sum_pairs(array, lower, upper):
+        """
+        Replace a pair of values in an array with their sum.
+
+        Args:
+            array (numpy.ndarray):
+                Array to be modified.
+            lower (int):
+                Lower index of pair.
+            upper (int):
+                Upper index of pair.
+
+        Returns:
+            numpy.ndarray:
+                Array where a pair of values has been replaced by their sum.
+        """
+        array[lower] = np.sum(array[lower : upper + 1])
+        return np.delete(array, upper)
+
+    def _combine_bin_pair_for_observation_frequency_monotonicity(
```

Comment:
Are you going for a record longest method name ever? :-)
If something wonderfully short and pithy comes to mind, please use it, but otherwise this will have to do.

Reply:
I've shortened this method name, so that it's less cumbersome.